### PR TITLE
Bug pylint 4019

### LINKED ISSUE #4019: inconsistent-return-statements with assert False in function call
Hello,

### Steps to reproduce
```
def f(x):
    if x == 1:
        return 42
    assert False
```

### Current behavior
```
example.py:2:0: R1710: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)
```

### Expected behavior

No error. This is pretty similar to this issue https://github.com/PyCQA/pylint/issues/1782, except that `assert False` is being used. I can currently work around with something like `raise RuntimeError('Unreachable')` but `assert False` is a common idiom to mark unreachable paths, and should be pretty easy to match.

### pylint --version output
```
pylint 2.6.0
astroid 2.4.2
Python 3.8.7 (default, Dec 24 2020, 17:53:09) 
[GCC 10.2.0]
```

### PR DESCRIPTION
**Pull request by hippo91**
## Steps

- [x] Add a ChangeLog entry describing what your PR does.
- [x] Write a good description on what the PR does.

## Description
During detection of `inconsistent-return-statements`, consider that an `assert False` statement is equivalent to a return node.

## Type of Changes

|   | Type |
| ------------- | ------------- |
| ✓  | :bug: Bug fix  |

## Related Issue
Closes #4019

### CODE REVIEW

**Review by Pierre-Sassoulas — APPROVED** [review]
Neat !

File: pylint/checkers/refactoring/refactoring_checker.py [hunk 1]
```diff
@@ -1381,6 +1381,13 @@ def _is_node_return_ended(self, node: astroid.node_classes.NodeNG) -> bool:
             return any(
                 self._is_node_return_ended(_child) for _child in all_but_handler
             ) and all(self._is_node_return_ended(_child) for _child in handlers)
+        if (
+            isinstance(node, astroid.Assert)
+            and isinstance(node.test, astroid.Const)
+            and not node.test.value
+        ):
+            #  consider assert False as a return node
```
Comment (cdce8p) [inline pylint/checkers/refactoring/refactoring_checker.py:1389]:
```suggestion
            # consider assert False as a return node
```

File: tests/functional/i/inconsistent/inconsistent_returns.py [hunk 2]
```diff
@@ -387,3 +387,21 @@ def bug_pylint_4122_bis(s):
         return n
     except ValueError:
         parser_error_name('parser error')
+
+
+def bug_pylint_4019(x):
```
Comment (cdce8p) [inline tests/functional/i/inconsistent/inconsistent_returns.py:392]:
```suggestion
# https://github.com/PyCQA/pylint/issues/4019
def bug_pylint_4019(x):
```
I like to add a direct link to the issue.

File: tests/functional/i/inconsistent/inconsistent_returns.txt [hunk 3]
```diff
@@ -15,3 +15,4 @@ inconsistent-return-statements:267:0:bug_3468:Either all return statements in a
 inconsistent-return-statements:277:0:bug_3468_variant:Either all return statements in a function should return an expression, or none of them should.
 inconsistent-return-statements:322:0:bug_pylint_3873_1:Either all return statements in a function should return an expression, or none of them should.
 inconsistent-return-statements:366:0:bug_pylint_4122_wrong:Either all return statements in a function should return an expression, or none of them should.
+inconsistent-return-statements:401:0:bug_pylint_4019_wrong:Either all return statements in a function should return an expression, or none of them should.
```
Comment (cdce8p) [inline tests/functional/i/inconsistent/inconsistent_returns.txt:18]:
```suggestion
inconsistent-return-statements:402:0:bug_pylint_4019_wrong:Either all return statements in a function should return an expression, or none of them should.
```
(Only if you agree with adding the link)

**Review by cdce8p — APPROVED** [review]
Just a few comments. Looks good overall!

File: pylint/checkers/refactoring/refactoring_checker.py (hunk 1 above)
Comment (Pierre-Sassoulas) [reply pylint/checkers/refactoring/refactoring_checker.py:1389]: https://github.com/psf/black/issues/2091 😄

Comment (Pierre-Sassoulas) [reply pylint/checkers/refactoring/refactoring_checker.py:1389]: @hippo91 just use ``setxkbmap -option "nbsp:none"`` in the meantime :p  https://stackoverflow.com/questions/31383074/how-to-disable-non-breaking-space-with-altgrspace?noredirect=1&lq=1

Comment (hippo91) [reply pylint/checkers/refactoring/refactoring_checker.py:1389]: Thanks to both of you!

File: tests/functional/i/inconsistent/inconsistent_returns.py (hunk 2 above)
Comment (hippo91) [reply tests/functional/i/inconsistent/inconsistent_returns.py:392]: Great idea!

File: tests/functional/i/inconsistent/inconsistent_returns.txt (hunk 3 above)
Comment (hippo91) [reply tests/functional/i/inconsistent/inconsistent_returns.txt:18]: :+1:

**Pull request merged**